### ISSUE TITLE
Is it possible to get a faster seekTo() response?

### ISSUE THREAD

**Issue by vxhviet** [Bilibili/ijkplayer#1247]
Hi *ijkplayer* developer(s), thank you for your excellent work.

I would like to use *ijkplayer* to seek through a local video via the `seekTo(long position)` method. The video is not actually played, but instead being scrubbed through [like this](https://www.youtube.com/watch?v=083iy30lbio&feature=youtu.be&t=1m21s).

I've tried both the `IjkMediaPlayer` and `IjkExoMediaPlayer` but they simply take quite sometime to decode the frame from the keyframe which means stuttering and slowness. `IjkMediaPlayer` perform slightly worse than `IjkExoMediaPlayer` in this matter.

Basically what I'm looking for is the [Input Seeking](https://trac.ffmpeg.org/wiki/Seeking) of ffmpeg and hopefully it will give me smoother scrubbing. Could you please confirm which ffmpeg seeking method that you used for `IjkMediaPlayer` and is there any way I can do to achieve a better performance?

Thank you so much for your time.

**Comment by bbcallen** [comment]
Actually, ijkplayer is tuned for poor network condition.
You can reduce the value of these options to reduce buffering time after seek.
https://github.com/Bilibili/ijkplayer/blob/master/ijkmedia/ijkplayer/ff_ffplay_options.h#L114

**Comment by vxhviet** [comment]
Hi @bbcallen, thanks a lot for your fast response.

Actually I'm developing on Android and this look like the options for the underlying ffmpeg. How can I modify these settings with the Java interface?

Sorry if I missed something obvious and thanks a lot for your hard work, cheers.

**Comment by bbcallen** [comment]
https://github.com/Bilibili/ijkplayer/blob/master/android/ijkplayer/ijkplayer-example/src/main/java/tv/danmaku/ijk/media/example/widget/media/IjkVideoView.java#L987

**Comment by vxhviet** [comment]
Dear @bbcallen, sorry for troubling you again.

I've tried tinkering with all those options in your suggestion. However they don't seem to have any effect on the seeking performance at all. I use them like this:

```java
mIjkMediaPlayer = new IjkMediaPlayer();
            mIjkMediaPlayer.setDisplay(holder);
            mIjkMediaPlayer.setDataSource(mVideoPath);
            
            mIjkMediaPlayer.setOption(IjkMediaPlayer.OPT_CATEGORY_PLAYER, "first-high-water-mark-ms", 10);
            mIjkMediaPlayer.setOption(IjkMediaPlayer.OPT_CATEGORY_PLAYER, "next-high-water-mark-ms", 10);
            mIjkMediaPlayer.setOption(IjkMediaPlayer.OPT_CATEGORY_PLAYER, "last-high-water-mark-ms", 10);
            /*mIjkMediaPlayer.setOption(IjkMediaPlayer.OPT_CATEGORY_PLAYER, "packet-buffering", 0);
            mIjkMediaPlayer.setOption(IjkMediaPlayer.OPT_CATEGORY_PLAYER, "max-buffer-size", 10);
            mIjkMediaPlayer.setOption(IjkMediaPlayer.OPT_CATEGORY_FORMAT, "fflags", "nobuffer");*/

            mIjkMediaPlayer.prepareAsync();
            mIjkMediaPlayer.setOnPreparedListener(new IMediaPlayer.OnPreparedListener() {
                @Override
                public void onPrepared(IMediaPlayer mp) {
                    mIjkMediaPlayer.seekTo(0);
                    mIjkMediaPlayer.pause();
                }
            });
```  

I've also tried the commented out part to no avail. Is there anything else I can try to achieve a better seeking performance?

**Comment by bbcallen** [comment]
Profile to see where it cost most.
